### PR TITLE
User feedback admin page

### PR DESCRIPTION
**Pull request by hansonchun-quartech**
### Description

This PR creates a user feedback page under admin that contains a data table of chat messages which have been rated by users. This table can be filtered and sorted based on search criteria. It can be filtered on `Feedback`, `Specialization`, `Start Date`, `End Date` and it can be sorted by `Date` and `Feedback`.

New `UserFeedback` models, controller, service, and a sort interface was added. The workflow first queries on the `chatmessages` container with given search and sort criteria to return a list of chat messages. Then the chat ids of these chat messages are further queried on `specialization` to return the final user feedback search result. Also a new UserFeedbackManager component was added to handle the front end functionality.

### Contribution Checklist

- [ ] The code builds clean without any errors or warnings
- [ ] The PR follows the [Contribution Guidelines](https://github.com/microsoft/chat-copilot/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/chat-copilot/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [ ] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

**Comment by GrahamS-Quartech** [comment]
Using cosmos as the storage provider, I get `Expression with NodeType 'Invoke' is not supported` when opening the user feedback page. I imagine this has to do with the And implementation calling Expression.Invoke in PredicateUtils.cs

**Comment by hansonchun-quartech** [comment]
> Using cosmos as the storage provider, I get `Expression with NodeType 'Invoke' is not supported` when opening the user feedback page. I imagine this has to do with the And implementation calling Expression.Invoke in PredicateUtils.cs

The issues integrating with CosmosDb should be fixed now. I also simplified the sorting part. There's not really an elegant way to handle sorting on multiple fields with Linq and CosmosDb, so I just made it to sort on a single field. I think that should be sufficient for our use cases.

File: webapi/Controllers/UserFeedbackController.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        ChatMessageRepository messageRepository
+    )
+    {
+        this._logger = logger;
+        this._userFeedbackService = new UserFeedbackService(sessionRepository, messageRepository);
+    }
+
+    [HttpPost]
+    [Route("userfeedback/search")]
+    public async Task<IActionResult> SearchUserFeedback([FromBody] UserFeedbackFilter filter)
```
Comment (GrahamS-Quartech) [inline webapi/Controllers/UserFeedbackController.cs:28]: I've always kinda felt that POST'ing for an operation like this isn't very RESTful but it's not super important.

File: webapp/src/components/chat/chat-list/ChatType.tsx [hunk 2]
```diff
@@ -63,19 +63,27 @@ export const ChatType: FC = () => {
...
                 dispatch(setIndexSelected(false));
+                dispatch(setUserFeedbackSelected(false));
             } else if (selectedAdminSubTab === 'indexes') {
                 dispatch(setAdminSelected(false));
                 dispatch(setIndexSelected(true));
+                dispatch(setUserFeedbackSelected(false));
+            } else if (selectedAdminSubTab === 'userFeedback') {
+                dispatch(setAdminSelected(false));
+                dispatch(setIndexSelected(false));
+                dispatch(setUserFeedbackSelected(true));
```
Comment (GrahamS-Quartech) [inline webapp/src/components/chat/chat-list/ChatType.tsx:80]: Following this established pattern is fine, but we ought to find a better way to handle this in a future ticket.

**Review by GrahamS-Quartech — DISMISSED** [review]

File: webapi/Controllers/UserFeedbackController.cs (hunk 1 above)
Comment (hansonchun-quartech) [reply webapi/Controllers/UserFeedbackController.cs:28]: Should I be using GET here instead?

File: webapp/src/components/chat/chat-list/ChatType.tsx (hunk 2 above)
Comment (hansonchun-quartech) [reply webapp/src/components/chat/chat-list/ChatType.tsx:80]: Yeah I'll make a ticket for this

File: webapi/Controllers/UserFeedbackController.cs (hunk 1 above)
Comment (GrahamS-Quartech) [reply webapi/Controllers/UserFeedbackController.cs:28]: IMO yes

**Review by GrahamS-Quartech — APPROVED** [review]

**Pull request merged**